src/tool_help.c

```diff
@@ -405,3 +407,82 @@ void tool_list_engines(void)
   curl_slist_free_all(engines);
   curl_easy_cleanup(curl);
 }
+
+/* Functions for outputting cheat-sheet argument. */
+/* Cheat sheet data structure organized by heading and value. */
+static const struct cheat_table {
+  const char *heading[2];
+} cheat_items[] = {
+  {{"Verbose", "-v, --trace-ascii file"}},
+  {{"Hide progress", "-s"}},
+  {{"extra info", "-w format"}},
+  {{"Write output", "-O, -o file"}},
+  {{"Timeout", "-m secs"}},
+  {{"POST", "-d string, -d @file"}},
+  {{"multipart", "-F name=value, -F name=@file"}},
+  {{"PUT", "-T file"}},
+  {{"HEAD", "-I"}},
+  {{"custom", "-X METHOD"}},
+  {{"Basic auth", "-u user:password"}},
+  {{"read cookies", "-b <file>"}},
+  {{"write cookies", "-c <file>"}},
+  {{"send cookies", "-b \"c=1; d=2\""}},
+  {{"user-agent", "-A string"}},
+  {{"Use proxy", "-x host:port"}},
+  {{"Headers add/remove", "-H \"name: value\", -H name:"}},
+  {{"follow redirs", "-L"}},
+  {{"gzip", "--compressed"}},
+  {{"insecure", "-k"}},
+  {{NULL, NULL}}
+};
+
+/* Support to print number of columns per screen width. */
+static void print_cheatsheet(int i, int col_num)
+{
+  /* Use consistent width for each column. */
+  int c;
+
+  for(c = 0; c < col_num; c++) {
+    /* The headers */
+    if(cheat_items[c + i].heading[0])
+      printf("%-30s ", cheat_items[c + i].heading[0]);
+    else {
+      col_num = c; /* redefine for uneven rows */
+      break;
+    }
+  }
+  puts("");
+  for(c = 0; c < col_num; c++)
+    printf("------------------------------ "); /* The separators */
+  puts("");
+  for(c = 0; c < col_num; c++) {
+    /* The options */
+    if(cheat_items[c + i].heading[1])
+      printf("%-30s ", cheat_items[c + i].heading[1]);
+  }
+  puts("\n");
+}
+
+/* Output cheat-sheet. */
+void tool_cheat_sheet(void)
+{
+  /* Get terminal width for proper formatting */
+  unsigned int cols = get_terminal_columns();
+
+  /* Get the table length. */
+  unsigned int table_length = sizeof(cheat_items) / sizeof(cheat_items[0]);
+
+  /* Handle for loop according to cols */
+  unsigned int i, j;
```

Comment:
```suggestion
  size_t cols = get_terminal_columns();
  size_t i, j;
```
drop comment saying trivia, drop unused (see below) table length.